pipeline/trusted-resources/pkg/trustedtask/trustedtask.go

```diff
@@ -0,0 +1,86 @@
+package trustedtask
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/apis"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+)
+
+type TrustedTask struct {
+	v1beta1.Task
+}
+
+func (t *TrustedTask) Validate(ctx context.Context) (errs *apis.FieldError) {
+	k8sClient := kubeclient.Get(ctx)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return apis.ErrGeneric(err.Error())
+	}
+	tektonClient, err := versioned.NewForConfig(config)
+	if err != nil {
+		return apis.ErrGeneric(err.Error())
+	}
+
+	if err := t.verifyTask(ctx, k8sClient, tektonClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *TrustedTask) verifyTask(
+	ctx context.Context,
+	k8sClient kubernetes.Interface,
+	tektonClient versioned.Interface) (errs *apis.FieldError) {
+	logger := logging.FromContext(ctx)
+	logger.Info("Verifying Task")
+
+	task := t.Task
+
+	cfg := config.FromContextOrDefaults(ctx)
+	cfg.FeatureFlags.EnableTektonOCIBundles = true
+	ctx = config.ToContext(ctx, cfg)
+
+	verifier, err := verifier(ctx)
+	if err != nil {
+		return apis.ErrGeneric(err.Error())
+	}
+
+	signature, err := getSignature(task.TaskMetadata())
+	if err != nil {
+		return apis.ErrGeneric(err.Error())
+	}
+
+	if err := VerifyInterface(ctx, task, verifier, signature); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getSignature fetches the signature specified in Task, Pipeline
+func getSignature(metadata metav1.ObjectMeta) ([]byte, error) {
+	sig, ok := metadata.Annotations[SignatureAnnotation]
+	if !ok {
+		return nil, fmt.Errorf("signature is missing")
+	}
+
+	signature, err := base64.StdEncoding.DecodeString(sig)
+	if err != nil {
+		return nil, err
+	}
+
+	delete(metadata.Annotations, SignatureAnnotation)
```

Comment:
This is going to modify the incoming object. Is this intentional? If so update the godoc, else let's remove this.

Reply:
bad me. think I should add a comment for this line - "remove the signature to compute the 'expected signature over the original/unsigned task"

refactored from these line: https://github.com/austinzhao-go/experimental/blob/feature/verify-task-pipeline-when-applying/pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go#L165 

but think now I found the problem - this delete will permanently delete the sig from original task, and  I should use a copy to do this deletion and move for the verification.